### PR TITLE
ARC-0053 Decentralized, Self-declared, & Verifiable Tokens, Collections, & Metadata

### PR DESCRIPTION
**Pull request by kylebeee**
Proposal for standardization of project / business self-declaration

### CODE REVIEW

**Comment by emg110** [comment]
Great idea! Although I 100% agree on proposed subject, about the ARC content and standard recommendations I have some questions:
1- Based on which global coverage or living standard this ARC is structured and proposed? Which living standard properties, declarations, versioning and formatting comply to and follow (Compliance and integration question)?

2- By which standard interfaces, gateways, or services in the real world this proposed ARC will be supported transparently without conversion, or interpretation (Usability and interoperability question)?

**Comment by kylebeee** [comment]
Hey!

1. This is a new ARC, I'm not aware of any previously established conventions for our ecosystem to share this kind of information. As stated in the motivation section for the ARC, we created it because there are a number of custom & centralized ways businesses we're doing simple parts of this. We saw an opportunity to create some decent standards that are relevant to almost every project or business that mints a token or NFTs. So we will definitely have to work to garner adoption with this standard. One of the ways we plan on doing that is introducing an open source app that formats these details & will upload to IPFS via a simple web form.

2. As far as adoption goes for using this information; projects & businesses would be free to utilize it as they see fit. Whats important to us is that the source information is trustworthy & clear in its purpose and i think this establishes that well.

**Comment by emg110** [comment]
Once again, great intentions, cause and initiative! But I did not get my answer because proposing standards IMHO should not be based on personal or organizational requirements or opinions when they are proposed for a larger scope community or in Algorand's case, global adoption!
What I meant was how this ARC guarantees interoperability and integration with already commonly used global living standards? Not expecting it to be globally adopted (I know this ARC is a new initiative)!

**Comment by kylebeee** [comment]
> What I meant was how this ARC guarantees interoperability and integration with already commonly used global living standards?

What "already commonly used global living standards" are you referring to here?

**Comment by emg110** [comment]
Exactly what I was hoping to hear from ARC proposer! I humbly think you need to have an answer for that question since you are proposing this ARC! 
Here are just some contextual (no direct relation to this subject but influential and on-topic examples) examples! In all these, you can find recommendations, structures, metadata or property names, and such, even including business declarations! What I meant was if you have a compliance strategy in this proposed ARC with such standards so that it does not lead to Algorand isolation and getting bubbled down with standards and rules that only make sense domestically!

- Schema.org: [Organization][LocalBusiness] [Corporation]...
- vCard and jCard
- Legal Entity Identifier (LEI)
- ISO/IEC 6523
- ISO 20022
- GS1's Global Data Dictionary (GDD)

**Comment by pbennett** [comment]
@emg110 
This is predominately about projects self-describing their NFT collections.  Each marketplace has to hand-define the collections at the moment and as I understand it, quite a mess.  Letting creators provide a way to define these themselves in an easily discoverable way seems a win.

**Comment by emg110** [comment]
Although I 100% agree with your comment @pbennett , I cannot understand what it has to do with my question and your bold disagreement with that! And As I said and need no re-assertion, I adore and agree with idea and was just asking if there are other standards being studied or complied to while proposing this one! Not disagree with it!

**Comment by kylebeee** [comment]
Those examples give me a much more tangible idea of what you we're thinking. As @pbennett said; this is primarily about collection declaration or minted assets and their grouping. I think those standards you're referencing take this in a bit of a different direction. There are escape hatches for extra information if someone were inclined to include those kinds of data. Perhaps in the future they could be brought into the fold if there was support & a clear desire for it.

That's really the reason this became an ARC. I think NFDs can work excellently for adding all kinds of data in a declarative and verifiable way. The main purpose starting out is sharing collections, tokens and basic information like the team & an FAQ. I have no doubt it will grow and i see this changing and being extended quite a bit. You have to start somewhere. If we were going off and creating our own custom standards that overlap with Schema.org and the others in their purpose i would agree it would have to be addressed and justified up front but thats not the case here.

**Comment by pbennett** [comment]
I think the title of 'project / business self-declaration' was a bit misleading and is probably part of emg's confusion.
Changing the title might make it more clear what the real intention is here ?

**Comment by SudoWeezy** [comment]
This one has a broader approach than https://github.com/algorandfoundation/ARCs/pull/122. 
It is worth gathering the community around an ARC meeting in the coming weeks to discuss this PR.

**Comment by SilentRhetoric** [comment]
As much as I like and support NFD, private entities do not belong in community standards.

I support the proposal, and, as an aside, I also think it might make sense to have a separate ARC for projects/businesses to declare info (unrelated to NFT collections).

**Comment by kylebeee** [comment]
Thanks for the feedback @SilentRhetoric, i realize people would like these standards to be completely agnostic but decided to push forward with it for a few reasons.

1. There is at the moment no other existing way to verify multiple wallets via smart contracts ( and readable state ) that i am aware of.

2. I thought this might be okay considering that in addition to being low cost they are also entirely in the control of the end user once they are minted.

3. We held a space discussing the ARC a bit ago and were given the okay as long as the ARC clarifies that it exists for NFDs specifically.

I am all for this being agnostic but an important aspect is this verifiability & I don't see someone creating parallel functionality to make this agnostic altruistically.

I want things to be as decentralized as they possibly can be and agree a private issuer for this identity-extension ARC is not ideal. With that being said I have a lot of faith in the NFD team and know they are working towards more decentralization for the NFD contracts daily.

I don't want to bar ourselves from creating useful things or supporting people & projects working to build on Algorand out of strict adherence to only moving forward when things are perfect.

**Comment by SudoWeezy** [comment]
What is more important than being platform agnostic (We always name IPFS, for example) is that we don't want a project from the ecosystem to have any advantages with a standard. 
However, I think we can move forward with the current version. It will be living anyway, and if/when a new domain name pops up, we can add a new part to this arc for the new one.

**Comment by joe-p** [comment]
I like the idea of not tying standards to specific projects. Making a generic contract that contains mapping of addresses to IPFS CIDs and only permitting that address to update it would be pretty straightforward.

**Comment by joe-p** [comment]
Just had a call with @kylebeee to discuss a bit more about why NFD was chosen to be included in the standard as it currently stand. The main thing NFDs offer that isn't available elsewhere is the ability to prove ownership of multiple accounts. If NFDs was not used and it was simply adding addresses to the JSON that indicated associated addresses, a project owner could add a wallet to the project that does not actually want to be added. 

I still like the idea of a more general standard and will explore what the effort of having an agnostic standard that supports this sort of address verification next week. I think there is also an opportunity to modularize the efforts here. Perhaps one standard for general linking of metadata to addresses, one standard for multi-address verification, and then ARC53 and other future can leverage those standards.

**Comment by kylebeee** [comment]
Great speaking with you @joe-p excited to see what comes out of this

**Comment by kylebeee** [comment]
Im happy to switch this to an agnostic standard with NFDs being the first to support the standard

**Comment by heartberg** [comment]
We are working on a fairly ambitious platform that among other things, enables Algorand projects to mint NFTs, associate these with a Collection and with the parent project. 

We were considering ARC-0012 as a viable solution for this, because:
- Information can remain mutable if needed (wallets, collection name, etc)
- It is a decentralized source of truth
We also have a mechanism to restrict which wallets can be associated with a project / collection. To avoid scam collections adding arbitrary influencer wallets to their metadata.

Our proposed platform does not directly compete with NFD as it doesn’t involve domains, but it does compete in certain use cases. Therefor it would not be in our interests to send users from our platform to the NFD platform. Our whole purpose is to enable projects to consolidate their utilities and assets in one platform.

If this proposed ARC is focused on solving the core problem, then we are all for it, but as it is described, it seems to be more focused on increasing NFD adoption. 
Even the demo link provided in the description immediately confronts the user with a big ‘Get domain’ button -https://akita.community/
No ARC should force a project to buy a proprietary product from a specific project like NFD.

**Comment by kylebeee** [comment]
Hi @heartberg, its not intended to spread adoption of NFDs. Its just that it was the only current multi-wallet verification system on mainnet. The core purpose of this is to trustlessly enable sharing collection and other metadata.

Im happy to go forward with it as an agnostic standard, the whole point of it being living was for it to change as new methodologies & use-cases arrive that meet that multi-wallet verification requirement.

**Comment by joe-p** [comment]
After talking with @pbennett I think accepting this ARC mostly as is and keep it as living is the best path forward. 

In an ideal world, NFDomains would have ABI interfaces that are published in an ARC and this standard would reference those interfaces, much like ENS. NFDomains, however, has been a project in the ecosystem for a long time and as such their contracts are fairly complex and don't leverage the ABI due to technical limitations that have historically (and some currently) exist with the AVM and ABI tooling. We could technically document the contracts as they work right now, but this would be a lot of work with not a ton of payoff. 

I think this means we should have this ARC exist explicitly referencing NFDomains, but with the ackowledgment that in the future the functionality used will be documented as interfaces rather than explicitly requiring usage of NFDomains.

Change I would recommend to @kylebeee before merging, assuming everyone is on-board with making this a living standard:

* Remove NFDomains from the ARC name
* Add a section that documents the fact that this ARC will eventually include interfaces (with NFDomains being the primarily implementation of those interfaces)

**Comment by cmd-cat** [comment]
I am not onboard with this ARC at all!
Referencing NFD in the ARC is detrimental to competition, and short sighted.

You have now heard from at least two voices in this thread that disagree with this proposal and who have stated they are working on projects that in part or full compete with NFD. You have also heard of an alternative more platform agnostic solution in ARC-0012.

Its really simple, we are working on an NFT project that will require a source of truth for collections. We do not want to send our users to NFD. We do not want our users to not use our service because they have to pay our fees and pay NFD fees. They can skip our service and just use NFD. End of our business.

Someone in this thread proposes making NFD the first implementation of this ARC, and you changed that to the ‘Primary’ implementation. Primary can be read to mean the ‘main’ implementation not the first implementation. I get the feeling that people have ulterior motives here to help structurally aid NFD’s business.

NFTs and tokens are the bedrock of any blockchain, having the ultimate source of truth for these point towards a single private business for which you need to buy their product is anticompetitive and ridiculous to even suggest.

Let’s build an ecosystem that is an equal playing field and not support one business over the other.
We should remove all mentions of NFD from this ARC, and ideally, we take another look at ARC-0012 instead :)

**Comment by SilentRhetoric** [comment]
Just to be clear, ARC-0012 is something entirely different than this.  Perhaps there is another ARC you have in mind as an alternative?

**Comment by joe-p** [comment]
Now that I think about it I think there is an opportunity to make a more general ARC for attaching metadata to an address. Simply have a contract that maps an address to a IPFS CAR. Within that CAR, there can be a file `arc53.json` that uses the JSON format described here and leverages NFDomains for the multi-address verification feature.

**Comment by cmd-cat** [comment]
@SilentRhetoric

Yes indeed. I meant ARC-0019
https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0019.md

**Comment by heartberg** [comment]
I agree with @cmd-cat in terms of removing any mentioning of NFD from the ARC.
ARC12 doesn't look suiting to this matter at all. Maybe because I used it in my previous post. 

The problem with ARC12 is that the data is stored on IPFS, which is not accessible by contracts and need an extra step for fetching the metadata.

@joe-p 's comment is good. Using a contract for storing the data and not IPFS would open up possibilities for contracts reading that metadata. Also storing it there via IPFS CARs is a good idea, loosing the contract reading feature though.

I would love to see this ARC to be standardized without any NFDs in mind. If you can have NFDs for verifying multi-addresses that's fine, but that could be another ARC, building on top of a standardized way for verifying Collection data etc.

**Comment by cmd-cat** [comment]
@kylebeee  When you say: “We’re removing mention of NFDs”, that means ANY mention right, not just removing the name from the title, but completely removing mention of NFD as a primary or initial or other implementation?

Just to cap this off, see this tweet:
https://twitter.com/NFDomains/status/1707818517167214958
NFD is promoting the Akita project; krby.algothen is building the Akita project; and retweeting the NFD tweet; and is also the author of this ARC proposal. There is likely coordination between the author of this proposal and NFD, and at the least a conflict of interest, which should have been a red flag to begin with. More generally we need to be really careful or forbid mention of proprietary products in ARCs, otherwise we risk disadvantaging competitors and sending the wrong message to developers about the openness of the Algorand platform.

We need generalized and open standards in Algorand, especially when it concerns the ultimate source of truth for “Tokens, (NFT) Collections, & Metadata” - pretty important stuff!

**Comment by kylebeee** [comment]
@cmd-cat I know the NFD team but am not affiliated in any capacity. Just a fan of the contracts capabilities. 

I'll remove mention of NFD in the title. I don't see an issue with listing the providers that are capable of supporting this ARC which for right now is just NFDs. 

I understand your concern, if your product/business builds the capabilities to support multi-wallet verification (which is crucial), I'll be happy to help push it as an official provider for this asap.

**Comment by SilentRhetoric** [comment]
Rather than including who can support the standard in the standard, it would be more appropriate to add this information to one of the ARC compatibility matrices, such as: [https://arc.algorand.foundation/nfts](https://arc.algorand.foundation/nfts).

**Comment by kylebeee** [comment]
Happy to do that instead, is that updated often?

Main concern is just that this needs to be easy to adopt and not listing providers forces the user/business to kinda put the pieces together themselves in some ways

**Comment by cmd-cat** [comment]
It seems that my concerns were legitimate, when you said: "We're removing mention of NFDs and moving forward with this", you were in fact just intending to remove NFD from the title, but still include NFD in the standard in some capacity. Sneaky.

ARCs are 'standards', they are not a list of your favorite providers. Once you include NFD in the standard, its likely to remain the primary or only name on the list. There is no guaranty that it will be updated in a timely way or done in a way that does not continue to emphasize NFD. I do not trust your intentions on this matter, due to your clear conflict of interests, your lack of candor in this thread, and the unusual amount of effort you are spending to promote NFD, while multiple people have disagreed with you.

I will also point out that another ARC exists related to NFT collections:
ARC-0030
https://github.com/algorandfoundation/ARCs/pull/122
In the discussion you will again see an NFD employee aggressively pushing for usage with NFDs.

You say that your main concern is the need for this to be easy to adopt. Engineers are smart, they can work out how best to implement this standard, if it is adopted. Your main concern should be whether this ARC distorts competition in the identity space, and whether it requires users to buy a propriety product to conform to the standard.

We need open, platform agnostic standards, not ones designed to explicitly benefit NFD.

**Comment by cmd-cat** [comment]
Another thing to add to this discussion: In addition to the airdrop that NFD performed, that promoted NFTs created by the author of this proposal, @kylebeee's NFD account shows that he currently owns 39 NF domains, has sold an additional 14 domains, and has 15 for sale right now (each for 10,000 Algo). And who know how many domains he has in other accounts.

Additionally, kylebeee runs a “domain club”, selling NF domains with over 50 holders:
https://twitter.com/kylebeeeee/status/1706377180476031049

It is crystal clear that @kylebeee is not looking out for the interests of developers as he claims, he is a domainer guy fighting hard to manipulate the ARC process to increase adoption of his preferred TLD, which benefits him personally.

Back in January, Exa_market held a poll to add support for another token – related to this, there was a very divisive discourse between the Akita army and other NFT community members. We really need to push back hard against these people who aggressively try to steer our ecosystem to serve their own interests above those of the whole Algofam!

**Comment by kylebeee** [comment]
@cmd-cat very clear you're trying to project quite a bit here.

I've only ever been open and honest about my motivations & reasoning with this ARC.

The ARC has already been changed to be agnostic, there is no mention of or requirement to use NFDs now.

Frankly the way you've been attempting to attack me is childish & unprofessional. I wish you luck with whatever product you're building and will be happy to help in terms of it supporting this ARC but i'd greatly appreciate if you communicated in good faith and stopped attempting to smear my intentions.

**Comment by kylebeee** [comment]
XGOV-156 is fulfilled & makes this standard much easier to adopt for both ecosystem builders & projects that need to share this information with those platforms.

[Watcher](https://github.com/kylebeee/arc53-watcher-go)
Is a block watcher & API server for the ARC53 standard, it indexes the data it finds on-chain that follows the ARC as well as updates it.

[Client](https://github.com/kylebeee/arc53-client-ts)
is a web app that makes it easy for projects to create their ARC53 JSON file for them to upload to IPFS & attach to their provider contract.a

You can use the client app [here](https://arc53-builder.vercel.app)

**Comment by dolphinkitty** [comment]
There is an alternative NFT Collection related ARC called ARC-40
https://github.com/algorandfoundation/ARCs/pull/280/files?short_path=e953fd2#diff-e953fd215d983cf6181e983b4d5f0684ed67bc8c82a87cc734d87a70b0fd1ee3

I asked Gemini AI, why is ARC-40 superior to ARC-53
Here is the response:

ARC-40 offers a significantly more robust and verifiable approach to NFT collections compared to ARC-53. While ARC-53 focuses on self-declaration and relies on potentially complex "provider" contracts for verification (which introduces significant ambiguity), ARC-40 establishes a standardized schema *and* a decentralized, platform-agnostic verification workflow.  Here's a breakdown of why ARC-40 is superior:

**ARC-53's Shortcomings:**

**Superficial "Verification":**  Generating a JSON file, even if it's hosted on IPFS and linked to a smart contract, doesn't inherently verify anything.  The ARC-53 app can create a JSON file with incomplete data.  ARC-53's reliance on "providers" for verification is vague and problematic.  How do these providers verify?  What are their standards? Who governs them?  If a provider verifies addresses, how does that prove their association with the *actual NFTs* in the collection?  ARC-53 provides no mechanism for linking entities to specific assets.  Furthermore, these "providers" would likely be platform-specific, reintroducing intermediaries and potentially undermining the decentralized nature of web3.

**Lack of Trust and Verify:**  Web3 is built on the principle of "trust and verify," not just "trust." ARC-53 asks us to trust the self-declaration without providing a robust way to verify the claims.  This is a significant weakness, especially in the context of NFTs where authenticity and provenance are paramount.

**Missing Minimum Requirements:**  ARC-53 states that only the `version` field is required. This effectively means *any* JSON blob can be considered a valid NFT collection declaration. This lack of structure and minimum requirements makes it practically useless for any meaningful verification or interoperability.  What constitutes a "valid" NFT collection under ARC-53?  It's unclear.

**Verification Scope:** ARC-53's verification, even if it worked as intended, seems limited to verifying wallets. It doesn't address the more critical issue of verifying the *association* between those wallets and the actual NFT assets.  This is a crucial gap.

**Future Incompatibility:** If ARC-53 were adopted, it's highly likely that a secondary ARC would eventually be necessary to standardize the verification flow. This makes ARC-53 an incomplete solution.

**ARC-40's Advantages:**

**Standardized Schema:** ARC-40 defines a clear and structured JSON schema for NFT collection metadata. This provides a consistent framework for describing collections, making it easier for platforms and tools to parse and interpret the data.

**Decentralized, Platform-Agnostic Verification Workflow:**  ARC-40's most significant advantage is its decentralized and platform-agnostic verification workflow.  The solution leverages Algorand's unique features, such as ARC-19 mutability and asset opt-in, enabling a unique and innovative approach to verification. Because it utilizes these native features, it does not rely on specific marketplaces or centralized services. This is crucial for interoperability and avoids vendor lock-in. It creates a system of bidirectional linkages between the Asset Set (collection), the Child Assets (NFTs), and the Contributors (creators, artists). This allows for trustless proof of control and attribution.

**Bidirectional Linkage:** The bidirectional linkage mechanism in ARC-40 is key. It ensures that the Asset Set metadata references the Child Assets, *and* the Child Assets reference the Asset Set.  This creates a strong, verifiable connection.  Similarly, contributors can be linked via their addresses or via special "contributor assets," which offer a novel approach to attribution.

**Clear Validation Rules:** ARC-40 defines explicit validation rules based on the `asset_type` (e.g., "collection" or "group"). This supports both verified and unverified collections to support different NFT use cases and provides clarity on what constitutes a valid NFT collection and how it can be verified.

**Flexibility and Extensibility:** ARC-40 is designed to be modular and extensible.  It can be adapted to different types of asset sets and use cases beyond just NFT collections.

**Contributor Attribution:** ARC-40 offers multiple ways to verify contributor associations, including opt-in and contributor assets. This is important for ensuring proper attribution and potentially facilitating decentralized royalty payments.

**Intellectual Property and Royalties:** ARC-40's schema includes fields like `license` and `royalties`, laying the groundwork for managing intellectual property rights and facilitating decentralized royalty payments.  While the actual distribution mechanism would require a separate implementation, ARC-40 provides the necessary metadata framework.

**Version Control:** ARC-40 incorporates version control mechanisms to track changes in the metadata over time, which is crucial for dynamic collections.

**In summary:** ARC-40 offers a comprehensive and verifiable approach to NFT collections, addressing the critical shortcomings of ARC-53.  ARC-40's standardized schema, decentralized and platform-agnostic verification workflow, and focus on bidirectional linkages make it a far superior solution for establishing trust and transparency in the NFT space.  It's a true "trust and verify" system, which aligns with the core principles of Web3, while also being platform agnostic.  To unlock the full potential of the maturing Algorand ecosystem, open standards are essential, ensuring a level playing field for new entrants to compete fairly and drive innovation.
